src/Shared/FileUtilities.cs

```diff
@@ -836,5 +837,36 @@ internal static string CombinePaths(string root, params string[] paths)
 
             return paths.Aggregate(root, Path.Combine);
         }
+
+        internal static string TrimTrailingSlashes(this string s)
+        {
+            return s.TrimEnd('/', '\\');
+        }
+
+        /// <summary>
+        /// Replace all backward slashes to forward slashes
+        /// </summary>
+        internal static string ToSlash(this string s)
+        {
+            return s.Replace('\\', '/');
+        }
+
+        internal static string NormalizeForPathComparison(this string s) => s.ToSlash().TrimTrailingSlashes();
+
+        internal static bool PathsEqual(string path1, string path2)
```

Comment:
It may not be too bad, though I do wish we had a better way to know. I'm certainly not suggesting rewriting it from scratch just for this hunch! I don't think we need to hold this PR for this concern, but we should definitely have this in the list of things to blame if there's a slowdown.